docs/RFCS/system_jobs.md

```diff
@@ -0,0 +1,424 @@
+initially equal to the `created` field and updated to the current timestamp
+whenever any change is made to the record. From these three fields, we can
+compute the duration of the job (`lastUpdated` - `started`) and the delay in
+starting the job (`started` - `created`).
+
+Every job that is `pending` will have a null `started` column, while every job
+that is not `pending` will have a non-null `started` column. It would be
+possible to instead use the same status for both `pending` and `running`,
+allowing the presence of the `started` column to distinguish between the two
+states, but the design as presented is more obvious.
+
+The type of job can be determined by reflection on the `details` oneof, which
+stores additional details relevant to a specific job type. The
+`BackupJobPayload`, for example, will store the target tables of the backup, so
+that the admin UI can visually flag tables that are currently being backed up.
+
+Finally, the `error` field stores the reason for failure, if any. This is the
+error message that is reported to the user through the normal query failure
+path, but is recorded in the table for posterity.
+
+Getting the fields in the `payload` protobuf right in this RFC is less critical
+than getting the columns in the `system.jobs` table right. In particular, the
+exact fields in the `SchemaChangeJobPayload` message are largely outside the
+scope of this RFC, because it is trivial to add missing fields later. The common
+fields in the `JobPayload` message are more important, because it's not entirely
+clear which fields belong in the table proper and which fields belong in the
+protobuf. See [Unresolved questions](#unresolved-questions) for more discussion
+on this point.
+
+## Expected queries
+
+To help evaluate which fields belong in the `payload` protobuf, following is a
+selection of SQL queries I expect to be run against the `system.jobs` table.
+Most of these queries will be executed by the database internals, though some
+are expected to be run manually by users monitoring job progress.
+
+To create a new job:
+
+```sql
+-- {} is imaginary syntax for a protobuf literal.
+INSERT
+  INTO system.jobs (jobType, status, payload)
+  VALUES ('backup', 'pending', {
+    query = 'BACKUP foodb TO barstorage',
+    creator = 'root',
+    created = now(),
+    last_updated = now()
+    backup_details = {
+      to = 'barstorage',
+      targets = ['foodb.tbla', 'foodb.tblb']
+    }
+  })
+  RETURNING id
+```
+
+To mark a job as running:
+
+```sql
+-- {...old, col = 'new-value' } is imaginary syntax for a protobuf literal that
+-- has the same values as old, except col is updated to 'new-value'.
+UPDATE system.jobs
+  SET status = 'running',
+      percentCompleted = 0,
+      payload = {...payload, started = now(), last_updated = now()}
+```
+
+To update the status of a running job:
+
+```sql
+UPDATE system.jobs
+  SET percentCompleted = 24.42,
+      payload = {...payload, last_updated = now()}
```

Comment:
Let's call it out though, because it's a pretty compelling reason to head more toward the "Narrow protobufs" alternative